src/index.js

```diff
@@ -5,22 +5,54 @@ import document from 'global/document'
 
 export default class ScrollListener extends PureComponent {
   state = {
-    scroll: 0
+    scroll: 0,
+    containerToListenOn: this.props.containerToListenOn || window,
+    containerToScroll: this.props.containerToScroll || document.body,
+    removeListener: this.props.removeListener
   }
 
   componentDidMount () {
-    this.setState({scroll: document.body.scrollTop})
-
-    window.addEventListener('scroll', this.recordPosition)
+    this.setState({scroll: this.state.containerToScroll.scrollTop})
+    this.attachListener()
   }
 
   componentWillUnMount () {
-    window.removeEventListener('scroll', this.recordPosition)
+    this.removeListener()
+  }
+
+  componentWillReceiveProps(nextProps, nextState) {
+    if (nextProps.removeListener) {
```

Comment:
hmmm i forgot about this. Basically we'll have to reconcile whether the productPage modal will always be mounted or not.  If it is always mounted, we'll have to figure out a way to mount/unmount this componentLib